shared/src/main/kotlin/org/javacs/kt/classpath/MavenClassPathResolver.kt

```diff
@@ -58,21 +59,34 @@ private fun mavenJarName(a: Artifact, source: Boolean) =
     if (source) "${a.artifact}-${a.version}-sources.jar"
     else "${a.artifact}-${a.version}.jar"
 
-private fun generateMavenDependencyList(pom: Path): Path {
-    val mavenOutput = Files.createTempFile("deps", ".txt")
-    val workingDirectory = pom.toAbsolutePath().parent.toFile()
-    val cmd = "$mvnCommand dependency:list -DincludeScope=test -DoutputFile=$mavenOutput"
-    LOG.info("Run {} in {}", cmd, workingDirectory)
-    val process = Runtime.getRuntime().exec(cmd, null, workingDirectory)
-
-    process.inputStream.bufferedReader().use { reader ->
+private fun readInputStream(process: Process, inputStream: InputStream) {
+    inputStream.bufferedReader().use { reader ->
         while (process.isAlive) {
             val line = reader.readLine()?.trim() ?: break
             if (line.isNotEmpty() && !line.startsWith("Progress")) {
                 LOG.info("Maven: {}", line)
             }
         }
     }
+}
+
+private fun generateMavenDependencyList(pom: Path): Path {
+    val mavenOutput = Files.createTempFile("deps", ".txt")
+    val workingDirectory = pom.toAbsolutePath().parent.toFile()
+    val cmd = "$mvnCommand dependency:list -DincludeScope=test -DoutputFile=$mavenOutput"
+    LOG.info("Run {} in {}", cmd, workingDirectory)
+    val process = Runtime.getRuntime().exec(cmd, null, workingDirectory)
+
+    val inputStream = Thread() { readInputStream(process, process.inputStream) }
+    val errorStream = Thread() { readInputStream(process, process.errorStream) }
```

Comment:
I am totally open to a more elegant way to write this thread/concurrency stuff. I'm no expert on Kotlin/the JVM...

Reply:
We could probably drop threads completely and simply do:

```kotlin
val process = Runtime.getRuntime().exec(cmd, null, workingDirectory)
process.waitFor()
```

Reply:
Alternatively, we could do:

```kotlin
val (result, errors) = execAndReadStdoutAndStderr(cmd, workingDirectory)
LOG.debug(result)
```

...analogously to how the `GradleClassPathResolver` does it. That way, we would still log the raw output for debugging purposes while not [potentially blocking forever](https://stackoverflow.com/a/5483953).